src/Orleans/Configuration/GlobalConfiguration.cs

```diff
@@ -415,7 +460,14 @@ internal GlobalConfiguration()
             ExpectedClusterSizeConfigValue = new ConfigValue<int>(DEFAULT_LIVENESS_EXPECTED_CLUSTER_SIZE, true);
             ServiceId = Guid.Empty;
             DeploymentId = Environment.UserName;
-            DataConnectionString = "";
+
+            DataConnectionStringForLiveness = "";
+            // Assume the ado invariant is for sql server storage if not explicitly specified
+            AdoInvariantForLiveness = AdoNetInvariants.InvariantNameSqlServer;
```

Comment:
This might become problematic if all of the ADO.NET specific code is separated to a csproj (and assembly) of its own.